### ISSUE TITLE
Large Dilated Convolution is working strange

### ISSUE THREAD

**Issue by reflelia** [PINTO0309/onnx2tf#251]
### Issue Type

Others

### onnx2tf version number

1.7.28

### onnx version number

1.12.0

### tensorflow version number

2.21.0r

### Download URL for ONNX

https://drive.google.com/file/d/11b9rL4tkDk1QyfeOjt-0OkYV_8PAmQB_/view?usp=sharing

### Parameter Replacement JSON

```json
{
    "format_version" : 1,
    "operations": [
        {
            "op_name": "Resize__19",
            "param_target": "outputs",
            "values": [128,128]
        }
    ]
}
```

### Description

1. Purpose : Personal development
2. What : In this model, I use Multi Dilated Convolution with stride=2 (x3 convolution) for extract features first. each convolutions have several dilation value(1, 2, 3). And the input shape is (b, 3, 256, 256), and if input tensor proceed 3 convolutions, I expect three (b, 3, 128, 128). but proceeded third convolution tensor's shape is (b, 3, 127, 127)
3. I try to resize the tensor(127, 127) to  (128, 128) by using json but its not working
4. I want to test this model in my android phone
5. None

**Comment by PINTO0309** [comment]
If a value of `>1` is specified for `dilations` and `strides` at the same time, there appears to be an error in the size of the trailing one pixel. The first three convolutions appear to produce large errors in the ONNX and TensorFlow output values, even though the output shapes are correctly `[1,128,128,3]`.

Note: This tool does not have the ability to extrapolate new OPs as specified by the user.

The figure below is a log of the process completed with a trial addition of zero padding when the final pixel is one pixel undersized as a work-around.
![image](https://user-images.githubusercontent.com/33194443/225607397-3b7997b0-15c1-4942-8e11-67e4ceb37acb.png)
　　:
　　:
![image](https://user-images.githubusercontent.com/33194443/225607476-989c520b-88d3-4eca-9d53-e2e9f73f3eb7.png)

Therefore, only `Conv_15` has a normal output, while `Conv_17` and `Conv_19` have abnormal output values.

The TensorFlow reference seems to say the following, but I don't know at the moment how it should be handled.

https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D
![image](https://user-images.githubusercontent.com/33194443/225609083-716f6770-46e2-48e6-bc11-b11d1c4a8518.png)

```
an integer or tuple/list of 2 integers, specifying the dilation rate to use for dilated convolution.
Can be a single integer to specify the same value for all spatial dimensions.
Currently, specifying any dilation_rate value != 1 is incompatible with specifying any stride value != 1.
```

**Comment by PINTO0309** [comment]
Note that the same result would be obtained even if the 1-pixel padding for fine adjustment is extrapolated before `Convolution`.

![image](https://user-images.githubusercontent.com/33194443/225617342-c0135771-1e0b-431f-b046-3b22bd5faa50.png)

![image](https://user-images.githubusercontent.com/33194443/225617072-24958ddb-b0f5-4927-b716-f057b77ad3aa.png)

**Comment by reflelia** [comment]
thank you!
Additionally, I need to convert the above model a few more times. Can you tell me how to add padding in tflite?

**Comment by PINTO0309** [comment]
Since it is open source, all you have to do is rewrite the source code and try it out. No merging will be done because of the problems with this test implementation.

https://github.com/PINTO0309/onnx2tf/blob/4a4f04757332c38f5fa55a7e1705c6a896920351/onnx2tf/ops/Conv.py#L521-L570

**Comment by reflelia** [comment]
I Tested the ported model, but in android, shape error occured because of [128, 128], [129, 129]
So I revised the 3 dilation, 3x3 kernel convolution to 2 dilation, 5x5 kernel convolution and it works
large Dilation value cause that bug
Thank you for advise

**Comment by PINTO0309** [comment]
I'm glad it seems to have worked. However, from your findings, I have a feeling that the problem is not so much the size of the `dilations`, but rather the fact that TensorFlow's internal implementation of rounding up and down fractions has different criteria than ONNX's, but I am not sure.

Incidentally, are you experiencing any degradation in the accuracy of the Float32 model you converted?
```
onnx2tf -i xxxx.onnx -cotof
```

**Comment by reflelia** [comment]
A converted model is not a fully trained model. I will complete the study later and let you know.

**Comment by PINTO0309** [comment]
>A converted model is not a fully trained model.

I did not explain it well enough. 

My tool has the ability to measure accuracy errors regardless of whether the model has been trained or not. It only checks all operations to see if the output of ONNX before conversion and the output of TensorFlow after conversion match or not. Thus, the `-cotof` option of onnx2tf is a simplified version that inputs meaningless all-1 values as dummy data and checks only for discrepancies in the output results of ONNX and TF.

Whether the model is trained or not is completely irrelevant.
